docs/x509/ocsp.rst

```diff
@@ -183,6 +201,137 @@ Loading Responses
         OCSPResponseStatus.UNAUTHORIZED
 
 
+Creating Responses
+~~~~~~~~~~~~~~~~~~
+
+.. class:: OCSPResponseBuilder
+
+    .. versionadded:: 2.4
+
+    This class is used to create :class:`~cryptography.x509.ocsp.OCSPResponse`
+    objects. You cannot set ``produced_at`` on OCSP responses at this time.
+    Instead the field is set to current UTC time when calling ``sign``.
+
+    .. note::
+
+        This builder is only used to create responses with the status
+        :attr:`~cryptography.x509.ocsp.OCSPResponseStatus.SUCCESSFUL`. All
+        other response statuses do not require signatures and have no payload.
+
+    .. method:: add_response(cert, issuer, algorithm, cert_status, this_update,
+                             next_update, revocation_time, revocation_reason)
+
+        This method adds status information about the certificate that was
+        requested to the response.
+
+        :param cert: The :class:`~cryptography.x509.Certificate` whose validity
+            is being checked.
+
+        :param issuer: The issuer :class:`~cryptography.x509.Certificate` of
+            the certificate that is being checked.
+
+        :param algorithm: A
+            :class:`~cryptography.hazmat.primitives.hashes.HashAlgorithm`
+            instance. For OCSP only
+            :class:`~cryptography.hazmat.primitives.hashes.SHA1`,
+            :class:`~cryptography.hazmat.primitives.hashes.SHA224`,
+            :class:`~cryptography.hazmat.primitives.hashes.SHA256`,
+            :class:`~cryptography.hazmat.primitives.hashes.SHA384`, and
+            :class:`~cryptography.hazmat.primitives.hashes.SHA512` are allowed.
+
+        :param cert_status: An item from the
+            :class:`~cryptography.x509.ocsp.OCSPCertStatus` enumeration.
+
+        :param this_update: A :class:`datetime.datetime` object (in UTC)
+            representing the most recent time at which the status being
+            indicated is known by the responder to be correct.
+
+        :param next_update: A :class:`datetime.datetime` object (in UTC) or
+            ``None``. The time at or before which newer information will be
+            available about the status of the certificate.
+
+        :param revocation_time: A :class:`datetime.datetime` object (in UTC) or
+            ``None`` if the ``cert`` is not revoked. The time at which the
+            certificate was revoked.
+
+        :param revocation_reason: An item from the
+            :class:`~cryptography.x509.ReasonFlags` enumeration or ``None`` if
+            the ``cert`` is not revoked.
+
+    .. method:: certificates(certs)
+
+        Add additional certificates that should be used to verify the
+        signature on the response. This is typically used when the responder
+        utilizes an OCSP delegate.
+
+        :param list certs: A list of :class:`~cryptography.x509.Certificate`
+            objects.
+
+    .. method:: responder_id(responder_cert, encoding)
```

Comment:
Is there a reason not to make this two methods?

Reply:
I think it's more understandable this way (to generate a responder_id on the response you provide the cert + the way you want it encoded). But it is mildly inconsistent with the OCSPResponse object itself, which has `responder_key_hash` and `responder_name` since we decided we didn't want a method that could return a byte string or an `x509.Name`.